docs/reddit-migration/strategy/schema.md

```diff
@@ -0,0 +1,61 @@
+# Establish a New Schema
+
+We want to establish a new schema so that we have a clean slate to work with and because there will be some incompatible changes with the current schema. An example of this is the `Post.post_id` column is a reference to the reddit id, but in the future needs to be the primary key of the row (see [Reddit Ids](reddit-ids.md)). We can't do that in-place with the current table.
+### Channels
+
+Channels should have the fields carried over with the following adjustments:
+
+ - `channel_id` -> `id`
+
+### Posts
+
+Posts should have the fields carried over with the following adjustments:
+
+ - `post_id` -> `id`
+ - `preview_text` / `plain_text` - we should to normalize this property / attribute down to a single field so we're not recomputing it every time a read occurs
+
+
+### Comments
+
+Comments should have the fields carried over with the following adjustments:
+
+ - `channel_id` -> `id`
+
+#### Comment Tree
+
+Prior art:
+
+ - SQL Antipatterns ("Naive Trees"):http://www.r-5.org/files/books/computers/languages/sql/style/Bill_Karwin-SQL_Antipatterns-EN.pdf
+ - Models for Hierarchical Data: https://www.slideshare.net/billkarwin/models-for-hierarchical-data
+ - Django treebeard: https://django-treebeard.readthedocs.io/en/latest/
+
+Reddit stores comment trees in Cassandra, which is overkill for our current use cases. That said, we need to be able to migrate the comments in such a manner that we achieve the following:
+
+- Maintain current API result structures
+  - Specifically, we need to support some form of reddit's "more comments" pagination. This **does not** need to partition the comments exactly as reddit would, the interface just needs to be compatible with how the frontend traverses it.
+- Support currently used sorting methods (best, old, new)
+- Support rendering an arbitrary subtree of the comments for the comment detail view
+
+From the Prior Art listed above, Materialized Paths are probably our best bet and we're already familiar to some degree with them because they're what `wagtail` uses under the hood for page hierarchies.
+
+The big thing here is that in order to support the sorts we need one for *each type of sort*. That precludes using a single table for everything so we'll want to have a hierarchical table for each sort and then `prefetch_related` on the `Comment` table so that code can work generically over similarly structured tables. This approach means these sorting hierarchy tables end up being an acceleration data structure for extracting a portion of the comment tree.
+
+### Votes
+
+We should have a table for `PostVote` and `CommentVote`, these should have the following columns:
```

Comment:
Would a `GenericForeignKey` to `Post`/`Comment` make sense here?

Reply:
I debated on that, I think the reason I settled for separate models is because I was thinking it'd be more efficient to perform aggregates over the table. It'd also be better for referential integrity.